pandas/core/arrays/string_.py

```diff
@@ -214,7 +215,11 @@ class BaseStringArray(ExtensionArray):
     Mixin class for StringArray, ArrowStringArray.
     """
 
-    pass
+    @doc(ExtensionArray.tolist)
+    def tolist(self):
+        if self.ndim > 1:
+            return [x.tolist() for x in self]
+        return list(self.to_numpy())
```

Comment:
Can we call a pyarrow specific function in case this is backed by a pyarrow table? And if yes, is that more performant?

Reply:
Yep, I tried the pyarrow `to_pylist` method but it seems to be quite a bit slower than going through numpy:

```
import pandas._testing as tm

arr = pd.array(tm.rands_array(10, 100_000), dtype="string[pyarrow]")

%timeit arr._data.to_pylist()
53.8 ms ± 473 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

%timeit list(arr._data.to_numpy())
5.26 ms ± 290 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

Reply:
yikes, thx for checking